components/layout_thread/lib.rs

```diff
@@ -1080,41 +1092,48 @@ impl LayoutThread {
 
         let element = match document.root_element() {
             None => {
+                let now = std_time::precise_time_ns();
+                if let ReflowGoal::LayoutQuery(_, time) = data.reflow_goal {
+                        self.layout_query_waiting_time = now - time;
```

Comment:
This is where we will want to update a histogram that's stored, rather than just a single numeric value. It probably makes sense to use a crate like [histogram](https://crates.io/crates/histogram).